tests/playwright/shiny/session/clientdata/test_clientdata.py

```diff
@@ -0,0 +1,25 @@
+import re
+
+from playwright.sync_api import Page
+
+from shiny.playwright import controller
+from shiny.run import ShinyAppProc
+
+
+def test_output_image_kitchen(page: Page, local_app: ShinyAppProc) -> None:
+
+    page.goto(local_app.url)
+
+    text = controller.OutputTextVerbatim(page, "clientdatatext")
+
+    # This doesn't cover all the clientdata values since some of them
+    # are environment-dependent. However, this at least checks that the
+    # clientdata object is available and that some values are present.
+    text.expect.to_contain_text("url_protocol = http")
+    text.expect.to_contain_text("url_pathname = /")
+    text.expect.to_contain_text(
+        re.compile("url_hostname = (localhost|127\\.0\\.0\\.1)")
+    )
```

Comment:
Could you test `url_search`, `url_hash_initial` and `url_hash` here? Those are all more valuable and likely to be used than, say, `url_protocol`.

Reply:
There's a shinycoreci app that already does a good job of covering the front end logic ([a30b8c8](https://github.com/posit-dev/py-shiny/pull/1832/commits/a30b8c8e3db70152aca103d3286e4e07e6ee9a37)), so I'd rather redo it